crates/orderbook/src/api/post_solver_competition.rs

```diff
@@ -0,0 +1,75 @@
+//! This is a private, undocumented api which will get replaced when we move the solution
+//! competition into the api.
+
+use crate::solver_competition::SolverCompetition;
+use anyhow::Result;
+use model::solver_competition::SolverCompetitionResponse;
+use reqwest::StatusCode;
+use std::{convert::Infallible, sync::Arc};
+use warp::{Filter, Rejection};
+
+fn request(
+) -> impl Filter<Extract = (String, u64, SolverCompetitionResponse), Error = Rejection> + Clone {
+    warp::post()
+        .and(warp::header::<String>("Authorization"))
+        .and(warp::path!("solver_competition" / u64))
+        .and(crate::api::extract_payload())
+}
+
+pub fn post(
+    handler: Arc<SolverCompetition>,
+    expected_auth: String,
```

Comment:
Instead of adding auth code - we can just disallow `POST solver_competition` at the ingress level. We already connect to the orderbook using a cluster-local URL (so doesn't go through the ingress) and makes it so we don't need to manage secrets.

Reply:
Not sure. Doing it only at the ingress level has the downside that an accidental misconfiguration could expose it. And we need to deal with another kubernetes configuration layer that I'm not familiar with. On the other hand it does make the code simpler and we don't want to keep this code for a long time anyway.

Reply:
Personally, I would prefer it at an ingress level (I tend to prefer having generic access control configured at an ingress level as Nginx does a really good job with this).

That being said, code is already there, simple enough, and will hopefully go away in the near future, so I am fine with including it in the PR.